azurelinuxagent/common/conf.py

```diff
@@ -127,7 +128,8 @@ def load_conf_from_file(conf_file_path, conf=__conf__):
     "ResourceDisk.MountPoint" : "/mnt/resource",
     "ResourceDisk.MountOptions" : None,
     "ResourceDisk.Filesystem" : "ext3",
-    "AutoUpdate.GAFamily" : "Prod"
+    "AutoUpdate.GAFamily" : "Prod",
```

Comment:
Have you considered making a distinct class for your appliance?  I think that would fit into the current model better.

Reply:
Can you elaborate, or point me at some documentation that describes the class model and the ability to define new ones / refine existing ones?

Reply:
Please provide a reference to documentation on the OS factory and instructions on defining a custom distribution.

Reply:
At this point, the documentation is the code.  I recommend you start with the [osutil](https://github.com/Azure/WALinuxAgent/tree/master/azurelinuxagent/common/osutil), and see how it is handled for the other flavors. Let me know if this does not make sense.